modules/pubxaiRtdProvider.js

```diff
@@ -80,31 +82,48 @@ export const setFloorsApiStatus = (status) => {
 
 export const getUrl = (provider) => {
   const { pubxId, endpoint } = deepAccess(provider, 'params');
-  return `${endpoint || FLOORS_END_POINT}?pubxId=${pubxId}&page=${
-    window.location.href
-  }`;
+  if (!endpoint) {
+    return null; // Indicate that no endpoint is provided
+  }
+  return `${endpoint || FLOORS_END_POINT}?pubxId=${pubxId}&page=${window.location.href}`;
 };
 
 export const fetchFloorRules = async (provider) => {
   return new Promise((resolve, reject) => {
     setFloorsApiStatus(FloorsApiStatus.IN_PROGRESS);
-    ajax(getUrl(provider), {
-      success: (responseText, response) => {
-        try {
-          if (response && response.response) {
-            const floorsResponse = JSON.parse(response.response);
-            resolve(floorsResponse);
-          } else {
-            resolve(null);
+    const url = getUrl(provider);
+    if (url) {
+      // Fetch from remote endpoint
+      ajax(url, {
+        success: (responseText, response) => {
+          try {
+            if (response && response.response) {
+              const floorsResponse = JSON.parse(response.response);
+              resolve(floorsResponse);
+            } else {
+              resolve(null);
+            }
+          } catch (error) {
+            reject(error);
           }
-        } catch (error) {
-          reject(error);
+        },
+        error: (responseText, response) => {
+          reject(response);
+        },
+      });
+    } else {
+      // Fetch from local storage
+      try {
+        const localData = storage.getDataFromLocalStorage('pubx:dynamicFloors') || window.__pubxDynamicFloors__;
```

Comment:
This data is written to localStorage by RTD module or will be written by the external floors loading script?

Reply:
yes